client/data/deposits/hooks.ts

```diff
@@ -24,11 +24,15 @@ export const useDeposit = (
 ): { deposit: CachedDeposit; isLoading: boolean } =>
 	useSelect(
 		( select ) => {
-			const { getDeposit, isResolving } = select( STORE_NAME );
+			const { getDeposit, isResolving, hasFinishedResolution } = select(
+				STORE_NAME
+			);
 
 			return {
 				deposit: getDeposit( id ),
-				isLoading: isResolving( 'getDeposit', [ id ] ),
+				isLoading:
+					! hasFinishedResolution( 'getDeposit', [ id ] ) ||
+					isResolving( 'getDeposit', [ id ] ),
 			};
```

Comment:
Adding `hasFinishedResolution` ensures that `isLoading` is `true` before the fetch request has begun.

Reply:
Note: `client/deposits/details/index.tsx` is the **only** consumer of the `useDeposit()` hook.